agent/consul/discoverychain/compile.go

```diff
@@ -739,6 +740,15 @@ func (c *compiler) getSplitterNode(sid structs.ServiceID) (*structs.DiscoveryGra
 			return nil, err
 		}
 		compiledSplit.NextNode = node.MapKey()
+
+		// There exists the possibility that a splitter may split between two distinct service names
+		// with distinct hash-based load balancer configs specified in their service resolvers.
+		// We cannot apply multiple hash policies to a splitter node's route action.
+		// Therefore, we attach the first hash-based load balancer config we encounter.
+		if !hasLB && node.LoadBalancer.IsHashBased() {
```

Comment:
Why is this restricted to hash-based only?

Reply:
It's because we only consider load balancing config from splitters when creating routes, and only hash-based policies get applied to route actions. Other LB policies only get applied to clusters.

If a non-hash-based policy was stored in a splitter it wouldn't have any effect: https://github.com/hashicorp/consul/blob/envoy-lb-v2/agent/xds/routes.go#L266-L275

This prefers storing a load balancing policy that the splitter might use. For example:
* `web` has a `least_request` policy
* `new-web` has a `ring_hash` policy

Splitter for `web` + `new-web` would store `new-web`'s LB config and apply its hash policies to the route action.

Maybe that's too much trickery though.